### PR TITLE
Bug 1773137 - New API endpoints needed for creating new components and editing current ones

### PR DESCRIPTION
**Pull request by dklawren**
This PR introduces new API endpoints to the current REST API for managing components for products. It covers getting a component, editing a component and creating a new component. This API is in the new versioned area for adding REST API functionality that everything else will eventually be migrated to. I had to change some other non-related code such as API login to fix some error handling issues found during testing. This PR came about due to others wanting to update the triage owner field of components. Rather than make a special endpoint just for doing that one edit, I thought better to go ahead and make a more robust way of editing components that needed to be done eventually anyway. Please let me know if you have any specific questions related to these changes.

### CODE REVIEW

File: docs/en/rst/api/core/v1/component.rst [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+name                      type     description
+========================  =======  =======================================================
+id                        int      An integer ID uniquely identifying the component in
+                                   this installation only.
+name                      string   The name of the component.
+description               string   A description of the component, which may contain HTML.
+is_active                 boolean  A boolean indicating if the component is active.
+default_bug_type          string   The default type for bugs filed under this component.
+default_assignee          string   The login of the default assignee for the component.
+default_qa_contact        string   The login of the default assignee for the component.
```
Comment (cgsheeh) [inline docs/en/rst/api/core/v1/component.rst:51]: `assignee` -> `QA contact`

File: qa/t/rest_triage_owners.t [hunk 2]
```diff
@@ -30,9 +30,6 @@ $t->get_ok(
   ->json_is('/Firefox/Installer/triage_owner', 'nobody@mozilla.org');
 
 my $data = $t->tx->res->json;
-use Bugzilla::Logging;
-use Mojo::Util qw(dumper);
-DEBUG(dumper $data);
```
Comment (cgsheeh) [inline qa/t/rest_triage_owners.t:35]: Unrelated?

**Review by cgsheeh — CHANGES_REQUESTED** [review]
All makes sense to me, just update some of the copy/pastes in the docs and this is okay to land. Also wondered about deletions in one file, if they're relevant can you explain why. Cheers!

File: qa/t/rest_triage_owners.t (hunk 2 above)
Comment (dklawren) [reply qa/t/rest_triage_owners.t:35]: Yeah but harmless. I left it in there accidentally as part of a previous merge. I can make it a separate PR if you really want.

Comment (cgsheeh) [reply qa/t/rest_triage_owners.t:35]: No need, just thought I'd ask.
